fix: changed to do redirects on server instead of client side [render preview]

SGI-CAPP-AT2 opened the pull request:
- before this redirects were made on client side which added extra page render
- now it'll send redirect to get request
- links shortified before this will not be affected but this change will not be available for them

closes: #2 

Results:
Right now if you receive GET request (Request sent by browser) to `links/r/:id` it'll send redirect to user so that the render of blank page will be avoided

Changes in nutshell:
![image](https://github.com/user-attachments/assets/7d63ebf4-60de-4501-b155-be967f5da5f0)

Screenshots:
Before (Still working): when we do GET to /r/ of client it'll always serve a blank page
![image](https://github.com/user-attachments/assets/05e837ac-241b-4e95-a478-1f00269e8e0a)

After: when we do GET to /r/ of server it'll redirect which will always show the desired web page instead of blank page
![image](https://github.com/user-attachments/assets/079998ad-7862-4dbd-b77c-1a7e03fc1f31)

SGI-CAPP-AT2 committed:
fix: changed to do redirects on server instead of client side

- before this redirects were made on client side which added extra page render
- now it'll send redirect to get request
- links shortified before this will not be affected but this change will not be available for them

SGI-CAPP-AT2 commented:
@mendacium-a11y , 
If anything is going wrong in the PR please let me know

mendacium-a11y commented:
can you add a video recording of the commit working locally

SGI-CAPP-AT2 commented:
@mendacium-a11y,

https://github.com/user-attachments/assets/5a1f46c9-c6e2-4188-b59a-f1a2dc4d724c

You can see that `[5173]/r/:id` shows some blank page and you can go to previous page
where as `[3000]/links/r/:id` directly redirects

SGI-CAPP-AT2 commented:
shortened urls after deploying this change might look like `shortify-ahj5.onrender.com/links/r/ididid`

mendacium-a11y commented on client/src/pages/Home.tsx:
lets deprecate the existing way to redirect, and just keep the server side one, with the same path of /r/:key

SGI-CAPP-AT2 replied:
yes, done

SGI-CAPP-AT2 committed:
deprecated client side handling

- previous commit migrated server side handling of shortened url
- now client side handling of same will be deprecated by this commit
- any previously generated urls are still available at new url under route `/links/r/:id`

SGI-CAPP-AT2 commented:
@mendacium-a11y done

SGI-CAPP-AT2 committed:
changed url handling get url to `/r` instead of `/links/r`
